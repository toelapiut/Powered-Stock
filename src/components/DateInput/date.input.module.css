.container{
    @apply flex flex-row  py-3 px-6 bg-navy rounded-lg
}

.start{

}

.start:nth-child(1){
    @apply pr-10 mr-10 border-r border-koamaru border-dashed
}

.label{
    @apply font-bold text-sm text-liberty leading-6
}

.date{
    @apply font-bold text-base text-white
}
