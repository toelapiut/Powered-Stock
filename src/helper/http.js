import axios from 'axios';

export const quandl = axios.create({
  baseURL: process.env.process.env.REACT_APP_BASE_URL,
  headers: {
    'Content-Type': 'application/json',
    'Accept': 'application/json',
    'Allow': 'GET, PUT, HEAD, OPTIONS'
  }
});

quandl.interceptors.request.use((config) => {
  config.params = config.params || {};
  config.params['key'] = process.env.REACT_APP_API_KEY;
  return config;
});
