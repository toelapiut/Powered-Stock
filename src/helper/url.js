export const url = {
  markets: 'markets',
  stocks: 'stocks/'
};