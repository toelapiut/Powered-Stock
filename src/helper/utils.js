import humps from 'humps';
import checkPropTypes from 'check-prop-types';
import {timeFormat} from 'd3-time-format';
import {bisector} from 'd3-array';
import moment from 'moment';

export const transpose = (rows, cols) => {
  /*
   * check if the row and col are empty and throw an error to
   * inform of the necessary requirements
   */
  if (typeof rows === 'undefined' || typeof cols === 'undefined') {
    throw new Error('row and col can not be undefined');
  }

  /*
   * throw error if the array isn't an array or col isn't an array
   */
  if (!Array.isArray(rows)) throw new Error('row must be  array');
  if (!Array.isArray(cols)) throw new Error('col must be  array');

  let dataset = [];

  for (let row of rows) {
    /*
     * check the data type before iterating through the matrix
     */
    if (!Array.isArray(row)) throw new Error('row must be a matrix row: [[a, b]]');
    let obj = {};
    row.forEach((item, index) => {
      obj[cols[index]['name']] = item;
    });
    dataset.push(obj);
  }

  return dataset;
};


export const transposeStock = (rows) => {
  let data = [];
  for (let row of rows) {
    let obj = {};
    const [date, close] = row;
    obj['date'] = date;
    obj['close'] = close;
    data.push(obj);
  }
  return data;
};

export const toCamelCase = (item) => {
  if (typeof item === 'undefined') {
    return item;
  }
  return humps.camelizeKeys(item);
};

export const checkPositivity = (num) => {
  if (typeof num === 'undefined') throw new Error('argument must not be empty');
  if (typeof num !== 'number') throw new Error('argument is not a number');
  let check = Math.sign(Number(num));
  return check !== -1;
};

export const formatDate = timeFormat('%b %d, \'%y');

export const getDate = (d) => new Date(d.date);

export const getStockValue = (d) => d.close;

export const bisectDate = bisector(d => new Date(d.date)).left;

export const getWindowDimensions = () => {
  const {innerWidth: width, innerHeight: height} = window;
  return {
    width,
    height
  };
};

export const dateFormat = (date, format = 'DD MMM, YYYY') => {
  return moment(date).format(format);
};

export const findTestByAttr = (component, attr) => {
  return component.find(`[data-test='${attr}']`);
};

export const checkProps = (component, expectedProps) => {
  checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
};

