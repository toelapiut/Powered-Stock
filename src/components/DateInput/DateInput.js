import React from 'react';
import PropTypes from 'prop-types';
import styles from './date.input.module.css';
import {Calendar} from '../Calendar/Calendar';
import {dateFormat} from '../../helper/utils';

export const DateInput = ({date, start, end, onOpenCalendar, isOpen, onChangeDates}) => {
  return (
    <div className={styles.container} data-test={'date-input-container'}>
      <button className={styles.wrap} onClick={onOpenCalendar}>
        <div className={styles.start}>
          <p className={styles.label}>Start date</p>
          <p className={styles.date}>{dateFormat(start)}</p>
        </div>
        <div className={styles.start}>
          <p className={styles.label}>End date</p>
          <p className={styles.date}>{dateFormat(end)}</p>
        </div>
      </button>
      {!isOpen && <div data-test={'toggle-calendar'}>
        <div className={styles.overlay} onClick={onOpenCalendar}/>
        <div className={styles.calendar}>
          <Calendar
            date={date}
            isOpen={isOpen}
            onChangeDates={onChangeDates}
            end={end}
            start={start}
          />
        </div>
      </div>
      }
    </div>
  );
};

DateInput.propTypes = {
  date: PropTypes.shape({
    start: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.object,
    ]),
    end: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.object,
    ])
  }),
  start: PropTypes.string,
  end: PropTypes.string,
  isOpen: PropTypes.bool.isRequired,
  onOpenCalendar: PropTypes.func.isRequired,
  onChangeDates: PropTypes.func.isRequired
};

