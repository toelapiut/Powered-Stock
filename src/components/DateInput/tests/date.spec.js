import React from 'react';
import DateInput from '../';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {findTestByAttr, checkProps} from '../../../helper/utils';


const setUp = (props = {}) => {
  return shallow(<DateInput {...props}/>);
};

let props = {
  date: {
    start: '01-04-2020',
    end: '01-04-2020'
  },
  name: 'Powered By People Inc',
  ticker: 'PBP',
  start: '01-04-2020',
  end: '01-04-2020',
  isOpen: false,
  onChangeDates: () => null,
  onOpenCalendar: () => null,
};

describe('DateInput Component', () => {
  describe('Component date inputs renders tests', () => {
    let component;
    beforeAll(() => {
      component = setUp(props);
    });

    test('should render snapshot correctly', () => {
      expect(toJson(component)).toMatchSnapshot();
    });

    it('should render without error', () => {
      const wrapper = findTestByAttr(component, 'date-input-container');
      expect(wrapper.length).toBe(1);
    });
  });

  describe('Check property types', () => {
    it('should not throw a warning', () => {
      const propErr = checkProps(DateInput, props);
      expect(propErr).toBeUndefined();
    });
  });


  describe('render without calendar', () => {
    let component;
    let calendarProps = {
      ...props,
      isOpen: true
    };
    beforeAll(() => {
      component = setUp(calendarProps);
    });

    it('should not render calendar component', () => {
      const wrapper = findTestByAttr(component, 'toggle-calendar');
      expect(wrapper.length).toBe(0);
    });
  });
});