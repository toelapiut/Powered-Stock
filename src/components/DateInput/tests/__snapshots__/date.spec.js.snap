// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateInput Component Component date inputs renders tests should render snapshot correctly 1`] = `
<div
  className="container"
  data-test="date-input-container"
>
  <button
    className="wrap"
    onClick={[Function]}
  >
    <div
      className="start"
    >
      <p
        className="label"
      >
        Start date
      </p>
      <p
        className="date"
      >
        04 Jan, 2020
      </p>
    </div>
    <div
      className="start"
    >
      <p
        className="label"
      >
        End date
      </p>
      <p
        className="date"
      >
        04 Jan, 2020
      </p>
    </div>
  </button>
  <div
    data-test="toggle-calendar"
  >
    <div
      className="overlay"
      onClick={[Function]}
    />
    <div
      className="calendar"
    >
      <Calendar
        date={
          Object {
            "end": "01-04-2020",
            "start": "01-04-2020",
          }
        }
        end="01-04-2020"
        isOpen={false}
        onChangeDates={[Function]}
        start="01-04-2020"
      />
    </div>
  </div>
</div>
`;
